src/main/java/com/cflint/plugins/core/PackageCaseChecker.java

```diff
@@ -43,47 +44,67 @@ public void expression(final CFExpression expression, final Context context, fin
     }
 
     private boolean checkComponentRegister(final Context context, final String componentPath, final String componentName) {
-        if (componentRegister.containsKey(componentName.toLowerCase())) {
-            List<String> filePathOfComponents = componentRegister.get(componentName.toLowerCase());
-            for (String filePathOfComponent : filePathOfComponents) {
-                if (filePathOfComponent.toLowerCase().endsWith(componentPath.toLowerCase())) {
-                    if (!filePathOfComponent.endsWith(componentPath)) {
-                        final String expectedPath = filePathOfComponent.substring(filePathOfComponent.length() - componentPath.length());
+        return checkComponentRegister(context, componentPath, componentName, false);
+    }
+
+    private boolean checkComponentRegister(final Context context, final String componentPath, final String componentName, Boolean fromExpressionRegister) {
+        String lowerComponentName = componentName.toLowerCase();
+        String lowerComponentPath = componentPath.toLowerCase();
+        if (componentRegister.containsKey(lowerComponentName)) {
+            HashSet<String[]> filePathOfComponents = componentRegister.get(lowerComponentName);
+            for (String[] filePathOfComponent : filePathOfComponents) {
+                if (filePathOfComponent[1].endsWith(lowerComponentPath)) {
+                    if (!filePathOfComponent[0].endsWith(componentPath)) {
+                        final String expectedPath = filePathOfComponent[0].substring(filePathOfComponent[0].length() - componentPath.length());
                         context.addMessage("PACKAGE_CASE_MISMATCH", expectedPath);
                     }
                     return true;
                 }
             }
         }
-        //otherwise remember the component use for when component is first registered.
-        final String key = componentName.toLowerCase();
-        if (!expressionCheckRegister.containsKey(key)) {
-            expressionCheckRegister.put(key, new ArrayList<PackageCaseCheckerEntry>());
+        if (!fromExpressionRegister) {
```

Comment:
The most important part of the fix, is here, to avoid injecting the same expression multiple times, the rest was just "optimization while trying to understand what was going on"

Reply:
Thanks for the hard work.   Looks good at a high level -and I'm trusting the unit tests :)